### PR TITLE
ci: Add GPU builds

### LINKED ISSUE #190: Add GPU builds in CI
We can't run them but we can at least check if the GPU build works

### PR DESCRIPTION
**Pull request by ddn0**
The CI won't pass until at least these issues are resolved:
- #195 
- #196 
- #194

(just for historical context, I noticed these errors while discussing #92 though they are not related to that issue itself)

Closes #190

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -134,6 +146,62 @@ jobs:
...
+            g++ \
+            gcc \
+            libboost-all-dev \
+            libeigen3-dev \
+            libopenmpi-dev \
+            llvm-7-dev \
+            openmpi-bin \
+            ssh
+
+          cmake -S . -B /tmp/build -DENABLE_DIST_GALOIS=ON -DENABLE_HETERO_GALOIS=ON
```
Comment (roshandathathri) [inline .circleci/config.yml:174]: -> cmake -S . -B /tmp/build -DGALOIS_ENABLE_DIST=ON -DGALOIS_ENABLE_GPU=ON

File: .circleci/config.yml [hunk 2]
```diff
@@ -134,6 +146,62 @@ jobs:
...
```
Comment (roshandathathri) [inline .circleci/config.yml:202]: -> cmake -S . -B /tmp/build -DGALOIS_ENABLE_DIST=ON -DGALOIS_ENABLE_GPU=ON

File: .gitmodules [hunk 3]
```diff
@@ -1,6 +1,3 @@
-[submodule "bliss"]
```
Comment (roshandathathri) [inline .gitmodules:1]: Why is this removed? I thought this was required for pangolin.

Comment (ddn0) [reply .gitmodules:1]: It is already vendored in external directly rather than as a submodule. The submodule itself was just something @chenxuhao made to track bliss upstream but it seems simpler to avoid inserting an intermediary.

File: .circleci/config.yml (hunk 2 above)
Comment (ddn0) [reply .circleci/config.yml:202]: Thanks for reminding me! Though this change only needed if this PR goes in after #199 . Otherwise it is actually correct as it stands.

File: .gitmodules (hunk 3 above)
Comment (roshandathathri) [reply .gitmodules:1]: Yes, it's better it's not a submodule but that is not related to this PR. As long as pangolin builds don't break, it's fine to be included in this PR.

Comment (ddn0) [reply .gitmodules:1]: FWIW, the reason the commit is here is because we have to enable submodules for GPU builds, so I took the opportunity to clean up our submodule configuration, which has these unused entries.

Comment (roshandathathri) [reply .gitmodules:1]: Ok, makes sense.

**Comment by roshandathathri** [comment]
LGTM

File: .travis.yml [hunk 4]
```diff
@@ -2,11 +2,8 @@ dist: bionic
 
 language: c++
 
-# At the moment, our LCI submodule has an additional submodule within it
```
Comment (insertinterestingnamehere) [inline .travis.yml:5]: Good catch. This comment is outdated now that the LCI submodule is gone.

**Comment by ddn0** [comment]
> It'll be great to have these working. Thanks for putting this together. One thing wasn't obvious to me: why do we need to check out submodules in all the builds when the submodules are really only used for the GPU code?

You are correct, but I figured it would be easier for maintenance and reduce the number of moving parts if we just standardized what it means to check out the repo (checked out or not vs checked out with submodules, checked out without ....). Alternatively, I imagine my future self to be a little surprised to see CI fail if I added a submodule that the CPU code needed, and my future self would have to add this same submodule checkout code anyway.
